pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/TeiXml.scala

```diff
@@ -2,13 +2,41 @@ package weco.pipeline.transformer.tei
 
 import grizzled.slf4j.Logging
 import weco.pipeline.transformer.result.Result
+import weco.pipeline.transformer.tei.transformers.{
+  TeiContributors,
+  TeiLanguages,
+  TeiNestedData
+}
 
 import scala.util.Try
 import scala.xml.{Elem, XML}
 class TeiXml(val xml: Elem) extends Logging {
   val id: String =
     getId.getOrElse(throw new RuntimeException(s"Could not find an id in XML!"))
 
+  lazy val scribesMap = TeiContributors.scribes(xml, id)
+
+  def parse: Result[TeiData] =
+    for {
+      summary <- summary
+      bNumber <- bNumber
+      title <- title
+      languageData <- TeiLanguages(xml)
+      (languages, languageNotes) = languageData
+      scribes <- scribesMap
```

Comment:
Two notes:

* For future!me/other reviewers: this is almost the same as the `parse()` function removed form TeiTransformer; the `scribes` part is the only change.
* Could we rearrange the order of the `for` comprehension so the order matches the order in which they're consumed by `TeiData`? Just to make the different parts easier to compare.

Reply:
> For future!me/other reviewers: this is almost the same as the parse() function removed form TeiTransformer; the scribes part is the only change



Just to add the reasoning for moving it here: the `scribes` function is a bit weird in that it returns a map and then each usage needs to know what to pick out of it. Also, because all scribes for all works in the tei are in the same place in the xml, it makes sense to parse it once and then pipe that map down when we're parsing nested data. But putting all of that in a `parse` function in `TeiTransformer` suddenly seemed to put too much knowledge about what needs to know what when parsing the Xml, and `TeiTransformer` didn't seem the right place anymore